tests/integration/targets/rds_cluster/tasks/create_cluster_vpc_sgs.yml

```diff
@@ -0,0 +1,205 @@
+      - "'endpoint' in _result_create_db_cluster"
+      - "'engine' in _result_create_db_cluster"
+      - _result_create_db_cluster.engine == "{{ engine }}"
+      - "'engine_mode' in _result_create_db_cluster"
+      - _result_create_db_cluster.engine_mode == "provisioned"
+      - "'engine_version' in _result_create_db_cluster"
+      - "'master_username' in _result_create_db_cluster"
+      - _result_create_db_cluster.master_username == "{{ username }}"
+      - "'port' in _result_create_db_cluster"
+      - "_result_create_db_cluster.port == {{ port }}"
+      - "'status' in _result_create_db_cluster"
+      - _result_create_db_cluster.status == 'available'
+      - _result_create_db_cluster.storage_encrypted == false
+      - "'tags' in _result_create_db_cluster"
+      - "'vpc_security_groups' in _result_create_db_cluster"
+      - "_result_create_db_cluster.vpc_security_groups | selectattr('status', 'in', ['active', 'adding']) | list | length == 3"
+
+  - name: Add a new security group without purge (test idempotence)
+    rds_cluster:
+      id: "{{ cluster_id }}"
+      state: present
+      vpc_security_group_ids:
+        - "{{ _result_create_sg.results.2.group_id }}"
+      apply_immediately: true
+      purge_security_groups: false
+    register: _result_create_db_cluster
+
+  - assert:
+      that:
+        - not _result_create_db_cluster.changed
+
+  - name: Add a security group with purge
+    rds_cluster:
+      id: "{{ cluster_id }}"
+      state: present
+      vpc_security_group_ids:
+        - "{{ _result_create_sg .results.2.group_id }}"
+      apply_immediately: true
+    register: _result_create_db_cluster
+
+  - assert:
+      that:
+        - _result_create_db_cluster.changed
+        - "_result_create_db_cluster.db_cluster_identifier == '{{ cluster_id }}'"
+        - "_result_create_db_cluster.vpc_security_groups | selectattr('status', 'in', ['active', 'adding']) | list | length == 1"
+
+  always:
+  - name: Remove security groups
+    ec2_group:
+      name: "{{ item }}"
+      description: "created by rds_cluster integration tests"
+      state: absent
+    loop: "{{ security_groups }}"
+
+  - name: Remove subnets
+    ec2_vpc_subnet:
+      cidr: "{{ item.cidr }}"
+      az: "{{ item.zone }}"
+      vpc_id: "{{ _result_create_vpc.vpc.id }}"
+      tags:
+        Name: "{{ resource_prefix }}-subnet"
+        Description: "Created by rds_cluster integration tests"
+      state: absent
+    ignore_errors: yes
+    loop: "{{ subnets }}"
+
+  - name: Delete VPC
+    ec2_vpc_net:
+      name: "{{ vpc_name }}"
+      state: absent
+      cidr_block: "{{ vpc_cidr }}"
+      tags:
+        Name: "{{ vpc_name }}"
+        Description: "Created by rds_cluster integration tests"
+    ignore_errors: yes
+
+  - name: Delete DB cluster without creating a final snapshot
+    rds_cluster:
+      state: absent
+      cluster_id: "{{ cluster_id }}"
+      skip_final_snapshot: True
+    ignore_errors: true
```

Comment:
security groups cannot removed here as long as the rds exists where it is used.  
same for vpc and subnet.  
At first, the rds must be removed. 2nd the security group, 3rd subnet and at last the vpc itself imo.

```suggestion
  - name: Delete DB cluster without creating a final snapshot
    rds_cluster:
      state: absent
      cluster_id: "{{ cluster_id }}"
      skip_final_snapshot: True
    ignore_errors: true
    
  - name: Remove security groups
    ec2_group:
      name: "{{ item }}"
      description: "created by rds_cluster integration tests"
      state: absent
    loop: "{{ security_groups }}"

  - name: Remove subnets
    ec2_vpc_subnet:
      cidr: "{{ item.cidr }}"
      az: "{{ item.zone }}"
      vpc_id: "{{ _result_create_vpc.vpc.id }}"
      tags:
        Name: "{{ resource_prefix }}-subnet"
        Description: "Created by rds_cluster integration tests"
      state: absent
    ignore_errors: yes
    loop: "{{ subnets }}"

  - name: Delete VPC
    ec2_vpc_net:
      name: "{{ vpc_name }}"
      state: absent
      cidr_block: "{{ vpc_cidr }}"
      tags:
        Name: "{{ vpc_name }}"
        Description: "Created by rds_cluster integration tests"
    ignore_errors: yes
```

Reply:
cool, duplicated ...